scripts/scil_resample_tractogram.py

```diff
@@ -0,0 +1,125 @@
+    std_group.add_argument('--streamline_wise_std', type=float,
+                           help='Noise to add to existing whole' +
+                                ' streamlines to generate new ones.')
+
+    sub_p = p.add_mutually_exclusive_group()
+    sub_p.add_argument('--gaussian', metavar='SIGMA', type=int,
+                       help='Sigma for smoothing. Use the value of surronding'
+                            ' X,Y,Z points on the streamline to blur the'
+                            ' streamlines.\nA good sigma choice would be '
+                            'around 5.')
+    sub_p.add_argument('--spline', nargs=2, metavar=('SIGMA', 'NB_CTRL_POINT'),
+                       type=int,
+                       help='Sigma for smoothing. Model each streamline as a '
+                            'spline.\nA good sigma choice would be around 5 '
+                            'and control point around 10.')
+
+    p.add_argument('-e', dest='error_rate', type=float, default=0.1,
+                   help='Maximum compression distance in mm after smoothing. '
+                        '[%(default)s]')
+    p.add_argument('--keep_invalid_streamlines', action='store_true',
+                   help='Keep invalid newly generated streamlines that may ' +
+                        'go out of the bounding box.')
+    p.add_argument('--seed', default=None, type=int,
+                   help='Use a specific random seed for the resampling.')
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    add_verbose_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if (args.point_wise_std is not None and args.point_wise_std <= 0) or \
+            (args.streamline_wise_std is not None and
+             args.streamline_wise_std <= 0):
+        parser.error('STD needs to be above 0.')
+
+    assert_inputs_exist(parser, args.in_tractogram)
+    assert_outputs_exist(parser, args, args.out_tractogram)
+
+    log_level = logging.WARNING
+    if args.verbose:
+        log_level = logging.DEBUG
+    logging.basicConfig(level=log_level)
+
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
+    original_number = len(sft.streamlines)
+
+    if args.nb_streamlines > original_number:
+        # Check is done here because it is not required if downsampling
+        if not args.point_wise_std and not args.streamline_wise_std:
+            parser.error("one of the arguments --point_wise_std " +
+                         "--streamline_wise_std is required")
+        sft = upsample_tractogram(
+            sft, args.nb_streamlines,
+            args.point_wise_std, args.streamline_wise_std,
+            args.gaussian, args.spline, args.seed)
+    elif args.nb_streamlines < original_number:
+        sft = get_subset_streamlines(sft, args.nb_streamlines, args.seed)
+    streamlines = compress_streamlines(
+        sft.streamlines, args.error_rate)
```

Comment:
I would not use compressing within this script.

Reply:
Yeah me neither df4d41a44f3f2bfd5fcf3c4899142cfeccbc5675